### PR TITLE
OADP-1833: Cleanup completed VSBs in backup finalize phase

### PR DESCRIPTION
**Pull request by shubham-pampattiwar**
`veleroImageFqin: 'quay.io/spampatt/velero:del-vsb-final-take-2'`

### CODE REVIEW

**Comment by shubham-pampattiwar** [comment]
`make update` runs successfully after updating the controller-gen but `make test` still fails :/

**Comment by kaovilai** [comment]
Then you have to update controllergen in the ci dockerfile or however that's run.

**Comment by sseago** [comment]
I have one concern here -- this involves CRD changes, so we will no longer be able to say that our Velero CRDs are identical to upstream. I don't know whether any of these differences will introduce incopatibilities or not, though.

File: hack/build-image/Dockerfile [hunk 1]
```diff
@@ -30,7 +30,7 @@ RUN wget --quiet https://github.com/kubernetes-sigs/kubebuilder/releases/downloa
     chmod +x /usr/local/kubebuilder/bin/kubebuilder
 
 # get controller-tools
-RUN go install sigs.k8s.io/controller-tools/cmd/controller-gen@v0.7.0
+RUN go install sigs.k8s.io/controller-tools/cmd/controller-gen@v0.12.0
```
Comment (kaovilai) [inline hack/build-image/Dockerfile]: build/ci-Dockerfile

**Comment by kaovilai** [comment]
Attempting to revert controller-gen changes.

**Comment by kaovilai** [comment]
This is the error in question. https://github.com/kubernetes-sigs/controller-tools/issues/613

**Comment by kaovilai** [comment]
Fix is in https://github.com/kubernetes-sigs/controller-tools/commit/32ad71090a629d8a633b74fc21cddc4a5f31ad8e which appears first in v0.9.0.
Also looks like I can't push to this PR.

**Comment by kaovilai** [comment]
This seems to work. Need a bit more work
https://github.com/shubham-pampattiwar/velero/pull/1

See Tests in https://github.com/openshift/velero/pull/270

**Review by kaovilai — APPROVED** [review]
lgtm

**Review by eemcmullan — APPROVED** [review]

**Comment by kaovilai** [comment]
@shubham-pampattiwar gotta merge this yourself whenever you're ready.. no OWNERS file = no auto merge.
Tho if you still want automerge we can remove the approved label requirement.. and rely on lgtm

**Comment by sseago** [comment]
Lets target this at oadp-1.2 directly (since we need to pull in some poentially-conflicting changes into konveyor-dev from upstream main, and then we may not need the overrides), and we can put this on konveyor-dev separately once that's done.

Otherwise, LGTM.

**Review by sseago — APPROVED** [review]
LGTM, but re-target at oadp-1.2 before merging.

**Comment by shubham-pampattiwar** [comment]
As discussed here is the PR for oadp-1.2: https://github.com/openshift/velero/pull/271

**Comment by sseago** [comment]
Closing this one, since oadp-1.2 one was merged and this will need rework before moving to konveyor-dev (post-next-rebase)

**Pull request closed without merging**